### PR TITLE
Display git build info (FINERACT-883)

### PR DESCRIPTION
**Pull request by xurror**
Display fineracts git build info actuator/info endpoint

## Description
Describe the changes made and why they were made. Ignore if these details are present on the associated Jira ticket

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [ ] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [ ] API documentation at https://github.com/apache/fineract/blob/develop/api-docs/apiLive.htm has been updated with details of any API changes.

- [ ] Integration tests have been created/updated for verifying the changes made.

- [ ] All Integrations tests are passing with the new commits.

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

### CODE REVIEW

**Comment by vorburger** [comment]
@xurror can you locally run and fix `./gradlew rat` ? Some license header is missing.

BTW this has nothing to do with [FINERACT-905](https://jira.apache.org/jira/browse/FINERACT-905), confusion.

PS please ignore the failed Build - Summary, and only look at Travis (see [FINERACT-907](https://jira.apache.org/jira/browse/FINERACT-907)).

**Comment by xurror** [comment]
> PS please ignore the failed Build - Summary, and only look at Travis (see FINERACT-907).

Was wondering where it came from. Thanks for the notice

> @xurror can you locally run and fix ./gradlew rat ? Some license header is missing.

Looking into it. It's kind of weird rat-report.txt is getting checked for license.

**Comment by xurror** [comment]
./gradlew rat  passes locally.

**Comment by xurror** [comment]
Oh now this interesting. Seems .git directory cannot be found here since it is a file which is automatically ignored

This also fails with docker build.

**Comment by xurror** [comment]
It's hard for me to figure out how to pull this off.

I'm thinking to set the option failOnNoDotGitFound to false. @vorburger, @awasum do you think we can go with that.

**Comment by nikpawar89** [comment]
@xurror  i saw the build, in your case following test case is failing: 
 org.apache.fineract.integrationtests.AccountingScenarioIntegrationTest > checkPeriodicAccrualAccountingFlow FAILED
    java.lang.AssertionError at AccountingScenarioIntegrationTest.java:588

For now may be you want to ignore the the testcase . But definitely it is not a long time solution.
It appears to me a concurrency issue on Loan entity object and is not limited to integration tests.

File: fineract-provider/build.gradle [hunk 1]
```diff
@@ -24,32 +24,33 @@ gradle clean tomcatrunwar
...
+    dependencies {
+        classpath 'io.spring.gradle:dependency-management-plugin:1.0.9.RELEASE'
+        classpath 'com.bmuschko:gradle-tomcat-plugin:2.5'
+        classpath 'nl.javadude.gradle.plugins:license-gradle-plugin:0.11.0'
+        classpath 'org.zeroturnaround:gradle-jrebel-plugin:1.1.2'
+        classpath 'org.springframework.boot:spring-boot-gradle-plugin:2.2.4.RELEASE'
+        // below
+        classpath "org.apache.openjpa:openjpa:$openJPAVersion"
+        classpath 'com.radcortez.gradle:openjpa-gradle-plugin:3.1.0'
+        classpath "org.nosphere.apache:creadur-rat-gradle:0.6.0"
```
Comment (vorburger) [inline fineract-provider/build.gradle]: This is interesting - why? Just newer version? And what does this have to do :smiling_imp: with FINERACT-883? Would you consider undoing this unrelated change here, and raising a separate (JIRA and?) PR for this?  Or does radle-git-properties somehow require this other RAT?

File: fineract-provider/build.gradle [hunk 2]
```diff
@@ -173,86 +175,85 @@ jacocoTestReport{
 }
 
 rat {
-  xmlOutput = false
```
Comment (vorburger) [inline fineract-provider/build.gradle:176]: as above, this to me seems totally unrelated to FINERACT-883 ?

File: fineract-provider/build.gradle [hunk 3]
```diff
@@ -403,7 +404,14 @@ war {
 
 license {
     header rootProject.file('../APACHE_LICENSETEXT.md')
-    excludes(["**/*.html", "**/*.mustache", "**/package-info.java", "**/keystore.jks", "**/swagger-ui/**"])
+    excludes([
```
Comment (vorburger) [inline fineract-provider/build.gradle:411]: Nice, but in a perfect world better raised in a very small separate PR. (I won't be a PITA and still merge it if you leave this one here though.  But that RAT related change really would be better to separate out.)

File: fineract-provider/build.gradle [hunk 4]
```diff
@@ -567,3 +576,11 @@ tasks.withType(SpotBugsTask) {
...
     reportLevel = "high"
 }
+
+gitProperties {
+    gitPropertiesResourceDir = "$buildDir/classes/java/main/resources"
+    dateFormat = "yyyy-MM-dd'T'HH:mmZ"
+    dateFormatTimeZone = "GMT"
+}
+
+generateGitProperties.outputs.upToDateWhen { false }
```
Comment (vorburger) [inline fineract-provider/build.gradle:592]: May be it would be useful to document what this does inline? According to https://github.com/n0mer/gradle-git-properties/blob/master/README.md#other-usages:

```suggestion
// make sure the generateGitProperties task always executes (even when git.properties is not changed)
generateGitProperties.outputs.upToDateWhen { false }
```

**Comment by vorburger** [comment]
@xurror re. test failure, follow #782 to find FINERACT-899 and note #774, I think if you rebase this, that will disappear.

> Oh now this interesting. Seems .git directory cannot be found here since it is a file which is automatically ignored

I don't think the `.git/` _is a file which is automatically ignored_, but I suspect there are several situations where it's not present... this project certainly should still build even if there is no `.git/` directory present.  This could happen if someone builds from the Release ZIP, and a few other situations I can imagine...

> I'm thinking to set the option failOnNoDotGitFound to false. @vorburger, @awasum do you think we can go with that.

... therefore something like that would probably be acceptable / good idea; see FINERACT-911.

However I do not find an option `failOnNoDotGitFound` so I suspect this was a typo?

**Review by vorburger — CHANGES_REQUESTED** [review]

**Comment by xurror** [comment]
> @xurror re. test failure, follow #782 to find [FINERACT-899](https://issues.apache.org/jira/browse/FINERACT-899) and note #774, I think if you rebase this, that will disappear.
> 
> > Oh now this interesting. Seems .git directory cannot be found here since it is a file which is automatically ignored
> 
> I don't think the `.git/` _is a file which is automatically ignored_, but I suspect there are several situations where it's not present... this project certainly should still build even if there is no `.git/` directory present. This could happen if someone builds from the Release ZIP, and a few other situations I can imagine...
> 
> > I'm thinking to set the option failOnNoDotGitFound to false. @vorburger, @awasum do you think we can go with that.
> 
> ... therefore something like that would probably be acceptable / good idea; see [FINERACT-911](https://issues.apache.org/jira/browse/FINERACT-911).
> 
> However I do not find an option `failOnNoDotGitFound` so I suspect this was a typo?

Oops, typo. It's failOnNoGitDirectory.

File: fineract-provider/build.gradle (hunk 3 above)
Comment (xurror) [reply fineract-provider/build.gradle:411]: Ok. Felt the line was getting long with what I added, so decided to break it.

**Comment by xurror** [comment]
@vorburger, this passes now. Hope the you OK with the actual changes

File: fineract-provider/build.gradle (hunk 3 above)
Comment (vorburger) [reply fineract-provider/build.gradle:411]: Sure, OK.

**Review by vorburger — APPROVED** [review]
LGTM!
